src/core/highlight.ts

```diff
@@ -0,0 +1,55 @@
+import { Token } from './token';
+import singleton, { type Prism } from './prism';
+import { stringify } from './stringify';
+import type { Grammar, GrammarToken, GrammarTokens, RegExpLike } from '../types';
+
+/**
+ * Low-level function, only use if you know what you’re doing. It accepts a string of text as input
+ * and the language definitions to use, and returns a string with the HTML produced.
+ *
+ * The following hooks will be run:
+ * 1. `before-tokenize`
+ * 2. `after-tokenize`
+ * 3. `wrap`: On each {@link Token}.
+ *
+ * @param text A string with the code to be highlighted.
+ * @param language The name of the language definition passed to `grammar`.
+ * @param options An object containing the tokens to use.
+ *
+ * Usually a language definition like `Prism.languages.markup`.
+ * @returns The highlighted HTML.
+ * @example
+ * Prism.highlight('var foo = true;', 'javascript');
+ */
+export function highlight (
+	this: Prism,
+	text: string,
+	language: string,
+	options?: HighlightOptions
+): string {
+	const prism = this ?? singleton;
+
+	const languageId = prism.components.resolveAlias(language);
+	const grammar = options?.grammar ?? prism.components.getLanguage(languageId);
```

Comment:
```suggestion
	const grammar = options.grammar ?? prism.components.getLanguage(languageId);
```

We don't need this if we accept the previous suggestion.